File: packages/app/frontend/src/services/lang-service.ts
import { LangService } from '@dailies-tracker/ui';
import * as app from '@/internal/main/App';

export async function langService(): Promise<LangService> {
  return {
    async getLang(): Promise<string> {
      return await app.GetLocale();
    },
    async setLang(lang: string): Promise<void> {
      await app.SetLocale(lang);
    },
  };
}

File: packages/app/lang-service.go
package main

import "github.com/spf13/viper"

type LangService struct{}

func (s *LangService) GetLang() string {
	return viper.GetString("locale")
}

func (s *LangService) SetLang(lang string) {
	viper.Set("locale", lang)
	viper.WriteConfig()
}
